maven-core/src/main/java/org/apache/maven/plugin/internal/ReadOnlyPluginParametersValidator.java

```diff
@@ -0,0 +1,80 @@
+package org.apache.maven.plugin.internal;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.plugin.descriptor.MojoDescriptor;
+import org.apache.maven.plugin.descriptor.Parameter;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluator;
+import org.codehaus.plexus.configuration.PlexusConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Print warnings if read-only parameters of plugin are used in configuration.
+ *
+ * @author Slawomir Jaranowski
+ */
+@Named
+@Singleton
+public class ReadOnlyPluginParametersValidator extends AbstractMavenPluginParametersValidator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger( ReadOnlyPluginParametersValidator.class );
+
+    @Override
+    protected Logger getLogger()
+    {
+        return LOGGER;
+    }
+
+    @Override
+    protected String getParameterLogReason( Parameter parameter )
+    {
+        return "is read-only, should not be used in configuration";
```

Comment:
Look at https://datatracker.ietf.org/doc/html/rfc2119 and should not, it gives me the feeling that the log message is wrong. As far as I understand: Setting those has no effect, but should not does not tell, no? If so, I would expect "shall not".

Reply:
unfortunately such settings have effect and parameters are changed according to user settings in configuration.

so `SHALL NOT` or even `MUST NOT` probably will be better.

Reply:
Are you going the change the message?

Reply:
Yes.
according to RFC:

> MUST NOT   This phrase, or the phrase "SHALL NOT", mean that the
   definition is an absolute prohibition of the specification.

I will choose  `MUST NOT` - I hope it will be the most accurate